code/wrap_label_input/README.md

```diff
@@ -0,0 +1,151 @@
+</td></tr>
+
+<tr><td>
+  Date
+</td><td>
+  2019-July-23
+</td></tr>
+
+<tr><td>
+  Explaination
+</td><td>
+   The flexbox makes it easy for elements to wrap to a new line. it also makes it easy for elements to grow according to the page size
+</td></tr>
+
+<tr><td>
+  Advantages
+</td><td>
+   This approach is not dependent on:
+   <ul>
+    <li>
+      media queries
+    </li>
+    <li>
+      complicated css containers.
+    </li>
+    <li>
+      libraries like bootstrap
+    </li>
+    <li>
+      tables
+    </li>
+   </ul>
+</td></tr>
+
+<tr><td>
+  Disadvantage
+</td><td>
+   <ul>
+    <li>
+      no use of media queries may be disadvantage if this is explicitly required
+    </li>
+    <li>
+      I (Tom) found the flexbox api personally very difficult. It is not clear when to use min-widt versus flex-basis etc. Also I find the flex short-hand very confusing.
+    </li>
+    <li>
+      When using table element one could get some of the functionality without hardcoding the max-width of the label.
+    </li>
+   </ul>
+</td></tr>
+
+<tr><td>
+  Alternatives
+</td><td>
+   <ul>
+    <li>
+      Libraries like bootstrap. They are not the standard and will outdate every 2 year.
+    </li>
+    <li>
+      Tables probably lack all sort of functionality like wrapping and scollable body.
+    </li>
+    <li>
+      grid. Not yet supported everywhere and might actually more complicated to do this with grid.
+    </li>
+    <li>
+      media query selectors. Creates more code and will be dependent on screensize (as opposed to parent element size) to work.
+    </li>
+   </ul>
+</td></tr>
+
+<tr><td>
+  Emerging technologies
+</td><td>
+   Grid is for now NOT supported everywere, but may do a good job for this as well.
+</td></tr>
+
+<tr><td>
+  Relation to our own architecture
+</td><td>
+   For our new apps we use flexbos alot and are no longer using CSS libraries like bootstrap.  
+   Therefore this seems a good fit.  
+</td></tr>
+
+</table>
+
+
+# Discussion
+I preferred writing out all flex properties instead of using the flex short-hand,   
+because I feel the flex short-hand is confusing.  
+Anyway, writing out all properties is confusing too.   
+Maybe we should just use whatever works.  
+
+I used CSS child selectors, but this might be done with simply adding classes as well.  
+Maybe this could be explained in another article under CSS / Selectors.  
+
+For now the flex-basis of the label is hardcoded in order for both items in the row to scale the same.  
+If flex-basis was not hardcoded then both labels would get a different width.  
+When using table elements this is not needed.  
+Tables align automaticly, but will not wrap or scroll.  
+
+Instead of using min-width, fles-basis is used.  
+The only reason (known by me) for this is that this simply works.  
+I do not understand the logic behind this and or if this is a bug.  
+
```

Comment:
I prefer using flex box for table than using table. Just a preference. 

I also normally use flex short-hand for flex-grow, flex-shrink and flex-basis. I think it is quite handful if we can understand it completely. So maybe we can have an open topic for this as well. 

When flex-basis is not set then item will take the width of the width of its content I suppose.

I have no idea in case of min-width.

Reply:
If it actually *is* a table, we should use table. Important for accessibility, screen readers and so on.

Reply:
I added my comments on this to the repo.
I think the discussion is interesting and we should probably investigate on this.